core/option.go

```diff
@@ -12,11 +12,13 @@ type params struct {
 	metrics            bool
 	chainID            string
 	availabilityWindow time.Duration
+	archival           bool
```

Comment:
Unfortunately, as `core` is technically the implicit `Availability` implementation for BN, it needs to be aware of storage status. 

FN + LN rely on their respective availability implementations to handle this.